modules/sdk-coin-near/src/near.ts

```diff
@@ -1077,4 +1068,43 @@ export class Near extends BaseCoin {
       }
     }
   }
+
+  /**
+   * Validates that the transaction matches what the user expects
+   */
+  private validateTokenEnablementTransaction(
+    transaction: Transaction,
+    explainedTx: TransactionExplanation,
+    txParams: VerifyTransactionOptions['txParams'],
+    txHex: string
+  ): void {
+    // Parse transaction from hex to validate actual transaction matches what user sees
+    const coinConfig = coins.get(this.getChain());
+    const freshTx = new Transaction(coinConfig);
+    freshTx.fromRawTransaction(txHex);
+    const freshTxData = freshTx.toJson();
+    const originalTxData = transaction.toJson();
+
+    // Verify key transaction fields match to prevent tampering
+    if (
+      freshTxData.signerId !== originalTxData.signerId ||
+      freshTxData.receiverId !== originalTxData.receiverId ||
+      freshTxData.publicKey !== originalTxData.publicKey ||
+      freshTxData.actions.length !== originalTxData.actions.length
+    ) {
+      throw new Error('Transaction hex does not match provided transaction');
+    }
+
+    // Validate addresses match between parameters and explained transaction
+    if (txParams.recipients && explainedTx.outputs) {
+      const expectedAddresses = txParams.recipients.map((r) => r.address);
+      const explainedAddresses = explainedTx.outputs.map((o) => o.address);
+
+      for (const addr of expectedAddresses) {
+        if (!explainedAddresses.includes(addr)) {
+          throw new Error(`Address mismatch: ${addr}`);
+        }
```

Comment:
Please move this snippet to a different validate/verifyFunction as the comments above, also, you can check if outputs matchs recipients on a single loop (O(n)):
```suggestion
   // remember to check that you're on a token enablement, except if this was checked for the general case or if this is always checked for other tx types. 
   if (txParams.recipients.length !== explainedTx.outputs.length) throw new Error('Error on token enablements: output doesn't match');
   
   const diffAddrs = explainedAddresses.filter((addr, index) => addr !== txParams.recipients[index].address);
   
   if (diffAddrs.length > 0) throw new Error(`Error on token enablements: address doesn't match for ${CONCAT your addresses code here}`);
```